### DISCUSSION TITLE
关于下一版本默认设置变动的意向调查

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#336] [mpv 通用]
当前版本2023V5的生命周期很长，我原计划是直接到年底。
下一版本发布前我打算看情况变动一些当前版本 [标准包的默认设置](https://github.com/hooke007/MPV_lazy/tree/e9495812086cf5984beed18ba8a8da8b4809678e)，因此收集一些反馈。（我不一定会改，你不提，一定不会改😀）

可以是关于以下方面的：

- mpv.conf中的选项
- input.conf中的快捷键是否缺乏常用的功能
- script-opts.conf中是否应该默认启用更多脚本功能选项
- 是否有其他你觉得很常用的着色器shader不在标准包中
- 以及其它我没想到的

>**Note**
`--vo=gpu-next` 相关的就不纳入考虑了，独有和差异化的东西还没能完全取代旧渲染器。
`--vo=gpu` 也依然能满足大多数需求长期使用，以后整体设置迁移时再做更改。

已在计划中的变动可查看 https://github.com/hooke007/MPV_lazy/commit/98d8439ef6fd04dda234d9207e06a369573ab9cb

**Comment by ghost** [comment]


**Reply by hooke007** [reply]
d3d11va-copy 在我常年的vs测试中偶尔有奇怪的随机丢帧问题，在我的amd独显上也有奇葩的效率问题（大概率驱动背锅）
d3d11va是不是还有win8+的系统需求，偶尔还有win7用户来反馈来着。。。
<s>目前的计划是 `auto-copy,nvdec-copy`</s>

<s>速度方面我打算追加方案，用 `shift+[` `shift+]` 的方式做逐步加减。</s>

**Reply by dyphire** [reply]
`auto-copy,nvdec-copy`其实等价于`auto-copy`，`auto-copy`本身就是一个优先级列表，会自行尝试回退解码。经过测试yuv444的视频`auto-copy`会自动从`d3d11va-copy`回退至`nvdec-copy`解码

**Reply by hooke007** [reply]
看来最近的提交使得 `auto` 也能fallback到那一堆copy上去了

**Reply by dyphire** [reply]
应该是了，我印象中之前`auto-copy`也不是这种行为

**Reply by natural-harmonia-gropius** [reply]
建议用 auto-safe-copy。

**Reply by dyphire** [reply]
`auto-safe-copy`是个不存在的参数，只有`auto-safe`和`auto-copy`
https://mpv.io/manual/master/#options-hwdec

**Reply by Lonely233233** [reply]
使用vlank是不是需要最新的驱动

**Reply by natural-harmonia-gropius** [reply]
> `auto-safe-copy` 是个不存在的参数

`auto-copy-safe` 这个是有的...

**Comment by 2607627233** [comment]
取消单文件控制？这个可以吗

**Reply by hooke007** [reply]
什么是单文件控制

**Reply by 2607627233** [reply]
取消script-opts.conf ，只保留文件夹

**Reply by hooke007** [reply]
理由是什么，优势是什么

**Reply by 2607627233** [reply]
感觉每次修改脚本设置都要对 script-opts.conf进行修改很繁琐（我的方式是直接修改xxx.conf，然后复制到script-opts.conf里），或许可以改成只对内置脚本进行集中设置？

**Reply by hooke007** [reply]
> 直接修改xxx.conf，然后复制到script-opts.conf里

我看不懂这操作啊，那为什么不直接在 script-opts.conf 里写。

而且我的算是脚本很少的整合了，抡上十几几十个脚本你是觉得每个脚本建立个conf方便还是只在一个文件里调所有设置方便？

另外可以看一下它的读取机制。你的需求甚至不怎么需要修改我的原文件的原始内容 https://github.com/hooke007/MPV_lazy/discussions/201

**Reply by 921j** [reply]
script-opts.conf中是否应该默认启用更多脚本功能选项  √

script-opts.conf 是最令人满意度的变动，特别是版本脚本做减法后，令每次升级需要迁移的变动项目少了很多！
我还巴不得它能更强大，能接管所有其他脚本配置才好，以后升级只需要保留它以及无脑复制自己定制的脚本及配置就行。
最近发现除了 input_uosc.conf 不能接管外，很多第三方的脚本配置也不能完美接管，比如 mpv_crop_script、contact_sheet、playlist_view，如果在 script-opts.conf 里改它们生成截图、缓存的目录，就会工作不正常。

**Reply by hooke007** [reply]
> 很多第三方的脚本配置也不能完美接管，比如 mpv_crop_script、contact_sheet、playlist_view，如果在 script-opts.conf 里改它们生成截图、缓存的目录，就会工作不正常。

给个具体示例我来测试能否复现。

**Comment by natural-harmonia-gropius** [comment]
~~替换 adaptive_sharpen 为 agyild/CAS~~
增加 hdeband

在 ssimd 的键位绑定里增加 correct-downscaling

**Reply by hooke007** [reply]
deband 目前是在profile里自动处理的。着色器的话不好配合
...也没有ssimd的键位（是不是看错片场了）

**Reply by natural-harmonia-gropius** [reply]
> 着色器的话不好配合

glsl-shaders-pre 应该没什么问题，hdeband 在同等平滑度下细节保留好太多了。

> 也没有ssimd的键位

我下的 noBin 里面有 ssimd……

https://github.com/hooke007/MPV_lazy/blob/c76fb0f6d717c784030e2274f01dbb1d61ce8881/portable_config/input_list.conf#L32

**Reply by hooke007** [reply]
glsl在默认的退出保存项目里。此外也很容易被ctrl+<kbd>`</kbd>清理掉

你下的不是我的吧（根本就不存在input_list这个文件）。main分支也不是懒人包

**Reply by natural-harmonia-gropius** [reply]
确实没有😓，不知道看混了什么

**Reply by dyphire** [reply]
我测试了下hdeband的表现，由于手上色带明显的片子不多，所以测试样本不足。
普通色带强度下hdeband表现优于mpv内置deband；中高色带强度下mpv内置deband通过调整`deband-iterations=2`和`deband-threshold=64`参数可以得到很好的去色带效果，hdeband虽然画面有部分更干净，但多次调参尝试后都无法做到彻底消除色带，有明显的色带残留，测试样本为Baha版本田小狼的暗景；烂色带强度下两者表现都一般

**Reply by dyphire** [reply]
只测试了真实（制作）的视频，而且主要偏向动画去色带的表现。演示片和8k片没有实际意义

**Reply by natural-harmonia-gropius** [reply]
https://www.youtube.com/watch?v=cRGwKuTqx0Y
这个用内置算法得到近似hdeband的效果时
https://www.youtube.com/watch?v=-xp0cWXJzuM&t=62s
这个楼梯内置算法很难保留住

**Reply by dyphire** [reply]
第一个视频和我测试的中高强度色带的样本相似，hdeband会残留色带无法消除（可能是我调参姿势不对
第二个视频62s处的白色楼梯真的会有区别吗？在我的测试中mpv内置deband和hdeband着色器开关都只影响了画面边缘的色带，楼梯部分没看出变化来

**Reply by natural-harmonia-gropius** [reply]
> 第一个视频和我测试的中高强度色带的样本相似，hdeband会残留色带无法消除（可能是我调参姿势不对

hdeband 需要放在 fsrcnnx、krig 这类应用于 luma 平面和 chroma 平面的缩放之前
手动添加要用 pre

> 第二个视频62s处的白色楼梯真的会有区别吗？在我的测试中mpv内置deband和hdeband着色器开关都只影响了画面边缘的色带，楼梯部分没看出变化来

测了一下我也看不出区别……属于是记忆错乱了

**Reply by dyphire** [reply]
> > 第一个视频和我测试的中高强度色带的样本相似，hdeband会残留色带无法消除（可能是我调参姿势不对
> 
> hdeband 需要放在 fsrcnnx、krig 这类应用于 luma 平面和 chroma 平面的缩放之前 手动添加要用 pre


用法是没错的：[dyphire/mpv-config/commit/5061dc](https://github.com/dyphire/mpv-config/commit/5061dcaa0f381b3b0f694494557f23f1099ba3b3)，主要对hdeband着色器的几个内置参数不太了解，怎么调参都去不干净中高强度的色带。mpv内置deband的预设面对中高强度的色带效果也一般，但可以通过调参得到很好的效果

**Comment by tanyiok1234** [comment]
其实可以有个RIFE-RealESRGAN的混合脚本，在我自己的4090 laptop上测试可以从1080p24fps -> 1080p48fps -> 2160p48fps，在140W-150W功耗左右不掉帧。对于台式机4080以上显卡等在番剧等应用场景来说还是很有价值的。


> import vapoursynth as vs
> from vapoursynth import core
> import k7sfunc as k7f
> 
> clip = video_in
> 
> 
> H_Pre = 1080
> Lt_D2k = True
> Sc_Mode = 1
> Fps_Num = 2
> T_Tta = False
> Ext_Proc = True
> Gpu = 0
> Gpu_T = 2
> St_Eng = False
> Ws_Size = 0
> Lk_Fmt = False
> 
> ret = k7f.FPS_CTRL(clip, fps_in=container_fps, fps_ret=True)
> clip = k7f.FMT_CTRL(clip, h_max=H_Pre, fmt_pix=1 if Lk_Fmt else 0)
> 
> clip = k7f.RIFE_NV(clip, lt_d2k=Lt_D2k, sc_mode=Sc_Mode, fps_num=Fps_Num, t_tta=T_Tta, ext_proc=Ext_Proc, gpu=Gpu, gpu_t=Gpu_T, st_eng=St_Eng, ws_size=Ws_Size)
> 
> Model = 5005
> Lt_Hd = True
> clip = k7f.ESRGAN_NV(clip, lt_hd=Lt_Hd, model=Model, scale=4 if Model==2 else 2, gpu=Gpu, gpu_t=Gpu_T, st_eng=St_Eng, ws_size=Ws_Size)
> 
> clip.set_output()

**Reply by hooke007** [reply]
vs的东西肯定不会考虑。k7sfunc已经是附带的东西了。论普适性，它自由构建的滤镜组合比rife+esrgan要实用的多的滤镜数不过来。这种超重需求自己搭建就行。配套文档也是齐全的

**Reply by tanyiok1234** [reply]
> vs的东西肯定不会考虑。k7sfunc已经是附带的东西了。论普适性，它自由构建的滤镜组合比rife+esrgan要实用的多的滤镜数不过来。这种超重需求自己搭建就行。配套文档也是齐全的

嘛，倒也没错，其实我是无所谓2333。之前只是觉得还要自己写一遍不够lazy，毕竟看个视频还要专门找播放器研究质量的人里说不定意外富哥挺多的（x

**Comment by Pr0pHesyer** [comment]
1.取消预设退出时记住着色器列表。
退出时记住着色器列表，那么mpv.conf里的着色器列表/着色器追加只在第一次生效，之后则会被saved-props.json中的覆盖而不生效。
修改mpv.conf以改变着色器是最直接也最符合直觉的，并且各文档都以mpv.conf为主，修改mpv.conf以改变应用选项。
2.修改文档及注释
若保留退出时记忆着色器列表，则希望能在mpv.conf里着色器部分添加注释，说明着色器列表默认在退出时记忆且会在应用启动时覆盖mpv.conf的选项。

**Reply by hooke007** [reply]
已采纳第二点

**Comment by verygoodlee** [comment]
新版本缩放相关的默认值有变化了，配置注释可能要更新一下了，性能要求比以前高 \
具体参考 https://mpv.io/manual/master/#gpu-renderer-options
```
scale=lanczos
# 不设定，默认与scale一致
cscale= 
# 新版本加入的全新算法，gpu-hq都没有改这个默认值，难道比mitchell还好吗
dscale=hermite

sigmoid-upscaling=yes
linear-downscaling=yes
correct-downscaling=yes
scaler-resizes-only=yes
dither-depth=auto
```
超低配旧机器可以开启`profile=fast`，基本可恢复旧版的配置
![image](https://github.com/hooke007/MPV_lazy/assets/45035465/301eebba-0f57-4e2b-9533-a0ade7c7d8ce)

另外想吐槽一下`profile=gpu-hq`，新版已改名`high-quality`，\
目前`gpu-hq`这个名字还能用，但是文档里已经搜不到`gpu-hq`了 \
`scale`和`csacle`都用上了`ewa_lanczossharp` \
把`csacle`开这么高，感觉性能浪费在这上面不太划算

**Reply by dyphire** [reply]
csacle 跟随 scale 纯属拍脑袋决定的，不建议参考官方配置
dscale 改为 hermite 就是图快还比 mitchell 锐，更推荐 catmull_rom 和 spline36 一些
默认 scale 算法改 lanczos，high-quality 配置使用 ewa_lanczossharp 后竟然不启用 scale-antiring。两个 ring 如此明显的算法也不针对性处理，那还不如改用 ewa_hanning

简单来说官方缩放算法相关变动很糟糕，别用（

**Reply by verygoodlee** [reply]
不是很爱折腾的普通用户，基本就默认缩放配置的，最多开个`gpu-hq`，结果这两个都挺糟糕的

**Reply by verygoodlee** [reply]
@dyphire 新版文档里`spline36`已经查无此人了，但是测试是还生效的，不知是文档里误删了还是打算将来移除

**Reply by dyphire** [reply]
> @dyphire 新版文档里 `spline36` 已经查无此人了，但是测试是还生效的，不知是文档里误删了还是打算将来移除

只是开发者认为默认值变了不再使用 spline36 所以没必要在文档里体现，见：https://github.com/mpv-player/mpv/commit/c934391e0730584b6a5512860c1da61962c4006f
mpv 有很多内部算法在官方手册里都是没有介绍的，比如我上面提的 ewa_hanning 算法，这是一个很优秀的 jinc 算法的变体，madvr 里的  jinc 算法实际上就是 ewa_hanning

**Reply by hooke007** [reply]
我目前的想法很简单，追着它改费劲。
让上游自己倒腾去，下次发新版的时候再合并这些所有改动。你们要追新版的注意追踪提交，我在长期内都不会同步。

**Reply by dyphire** [reply]
> 我目前的想法很简单，追着它改费劲。 让上游自己倒腾去，下次发新版的时候再合并这些所有改动。你们要追新版的注意追踪提交，我在长期内都不会同步。

下个版本你要考虑切换到 --vo=gpu-next 了，mpv 最近改动默认值开始无视对 --vo=gpu 的影响了。比如 https://github.com/mpv-player/mpv/commit/e7bd330ed073afe973dc9bdae669ac4a4a0017f2 这个提交改动的 hdr 峰值检测的默认值在 --vo=gpu 下很糟糕（gpu 在线性空间下进行峰值检测，gpu-next 在 PQ 空间下进行峰值检测）

**Reply by dyphire** [reply]
我个人也不建议频繁追踪 mpv 变动，可以看看我仓库的提交历史：[dyphire/mpv-config/commits](https://github.com/dyphire/mpv-config/commits/master/)
短短几天同步 mpv 上游变动了三次（
真的对 mpv 变动比较关心的话，可以时不时看下我仓库相应的提交（后续应该也会降低跟踪频率

**Reply by verygoodlee** [reply]
> 默认 scale 算法改 lanczos，high-quality 配置使用 ewa_lanczossharp 后竟然不启用 scale-antiring

`scale-antiring`貌似对`ewa_`系列不生效，https://mpv.io/manual/master/#options-scale-antiring
`Note that this doesn't affect the special filters bilinear and bicubic_fast, nor does it affect any polar (EWA) scalers.`

`ewa_lanczossharp`带来的`ring`估计官方觉得可以接受，更锐利的`ewa_lanczos4sharpest` 官方推荐搭配anti-ringing着色器使用

**Reply by dyphire** [reply]
`ewa_`系列算法要启用`scale-antiring`效果需使用--vo=gpu-next 的 vo 后端，--vo=gpu 已不再维护新特性
`ewa_lanczos4sharpest`一坨 shit，别用。`ewa_lanczossharp`不搭配`scale-antiring`不推荐使用，只有`ewa_hanning`的 ring 比较少可以不用`scale-antiring`以保持锐利度

**Reply by snylonue** [reply]
> 默认 scale 算法改 lanczos，high-quality 配置使用 ewa_lanczossharp 后竟然不启用 scale-antiring。两个 ring 如此明显的算法也不针对性处理，那还不如改用 ewa_hanning

vcb 教程里推荐的就是 ewa_lanczossharp，我还用了挺久的(后来 gpu_next 把它移除以后就换了 ewa_lanczos)

**Reply by hooke007** [reply]
gpunext没有真正移除它，因为本来就比gpu的版本sharp，只是没有必要保留无意义的类似重复项。这点在官方wiki中有记录

**Reply by 2607627233** [reply]
`profile=fast `
`profile=high-quality`
这俩我用起来完全没区别，指功耗上

**Comment by piaoying9** [comment]
希望集成历史播放列表的脚本[simplehistory.lua](https://github.com/dyphire/mpv-config/blob/master/scripts/simplehistory.lua)，这个应该是比较常用的功能

**Reply by hooke007** [reply]
除非与uosc高度集成，否则不会新增任何复杂配置的脚本

**Reply by dyphire** [reply]
这个脚本虽然我也有用，但配置太过复杂了，这开发者的 mpv 脚本都有这个问题。
我也不推荐集成它，如果要集成历史记录脚本，更推荐这个：[recent-menu](https://github.com/natural-harmonia-gropius/recent-menu)

**Reply by piaoying9** [reply]
尝试了一下[recent-menu](https://github.com/natural-harmonia-gropius/recent-menu)，确实好用，也与uosc高度集成，感觉很合适。

**Comment by gdm257** [comment]
考虑下改为窗口等比例缩放吗？

mpv 原版默认就是，这样窗口缩放时不会出现黑边。没记错 MPC/Potplayer 默认也是等比例，基本符合大众习惯

```
# mpv.conf
keepaspect-window = yes
```

**Reply by hooke007** [reply]
不。mpv的默认窗口缩放才是特立独行的那一个

**Reply by dyphire** [reply]
~~你搞错了吧，mpv 的 hidpi-window-scale 缩放行为是特立独行的（我个人挺喜欢~~
~~keepaspect-window 窗口比例随视频比例变化，这是大多数播放器都有的行为，mpc-hc/be 也没什么不同。锁定窗口比例加黑边才是比较特立独行的（mpc 和 mpv 都可以通过设置实现~~

Edit: 是我整错了，把 auto-window-resize 和 keepaspect-window 搞混了，keepaspect-window 确实和其他播放器行为不同

**Reply by gdm257** [reply]
重置了下还真是，太久不用都忘了（喜欢+1
